### PR TITLE
Go implementation of rewriteBuffer

### PR DESCRIPTION
**Pull request by dakimura**
Hi @Notargets san,
in order to improve the maintainability of marketstore by unifying its language, I implemented a Go implementation of RewriteBuffer and compared the performance between C and Go implementations by running a benchmark test.

The following result shows that there is not a significant difference between them on latency or allocated memory.
```
dakimura:~/projects/go/src/github.com/dakimura/marketstore/akimura (go-rewrite-buffer *+$) $ go test -bench . -benchmem
goos: darwin
goarch: amd64
pkg: github.com/alpacahq/marketstore/akimura
Benchmark_TestRewriteBuffer_Go-8         8249734               143 ns/op              80 B/op          1 allocs/op
Benchmark_TestRewriteBuffer_C-8         10489608               115 ns/op              80 B/op          1 allocs/op
```

Could you review it and give me your opinions?
Thank you!

### CODE REVIEW

File: executor/readvariable.go [hunk 1]
```diff
@@ -92,12 +90,8 @@ func (r *reader) readSecondStage(bufMeta []bufferMeta, limitCount int32, directi
 					numVarRecords = numberLeftToRead
 				}
 			}
-			rbTemp := make([]byte, numVarRecords*(varRecLen+8)) // Add the extra space for epoch
-
-			arg1 := (*C.char)(unsafe.Pointer(&buffer[0]))
-			arg4 := (*C.char)(unsafe.Pointer(&rbTemp[0]))
-			C.rewriteBuffer(arg1, C.int(varRecLen), C.int(numVarRecords), arg4,
-				C.int64_t(md.Intervals), C.int64_t(intervalStartEpoch))
+			rbTemp := RewriteBuffer_Go(buffer,
```
Comment (dakimura) [inline executor/readvariable.go]: change from C implementation to Go implementation

File: executor/rewritebuffer.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package executor
```
Comment (dakimura) [inline executor/rewritebuffer.go]: Go implementation of Rewrite Buffer

File: executor/rewritebuffer_test.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
+package executor
```
Comment (dakimura) [inline executor/rewritebuffer_test.go]: compatibility & benchmark test for both implementations

**Comment by umitanuki** [comment]
I like the idea and I believe the same performance can be achieved in Go. Once thing I wanted to do is to completely avoid cgo/gcc to build the source, since the dependency is currently propagated to the downstream client programs too. Could you separate the cgo stuff from the main build, if it is only for the comparison purpose?

**Comment by dakimura** [comment]
> Could you separate the cgo stuff from the main build, if it is only for the comparison purpose?

Sure thing!
marketstore has 2 more C implementations: `timsort.c` and `quicksort.c` . 
Let me create another pull request to replace them to Go implementations and completely remove 
C implementations and its build process from the codebase!

**Comment by dakimura** [comment]
Please see https://github.com/alpacahq/marketstore/pull/281 .

**Pull request closed without merging**